userspace/libsinsp/mpsc_priority_queue.h

```diff
@@ -155,6 +155,23 @@ class mpsc_priority_queue
 		}
 	}
 
+	/**
+	 * @brief Sets the maximum capacity of the queue. Returns false
+	 * if the the specified capacity cannot be set (when the current queue
+	 * size is bigger than the specified capacity). 
+	 */
+	inline bool set_capacity(size_t capacity)
+	{
+		std::scoped_lock<Mtx> lk(m_mtx);
+		if(m_queue.size() <= capacity)
+		{
+			m_capacity = capacity;
```

Comment:
Am I missing something here or why do we not have a call `m_queue.set_capacity` to actually change the capacity? Or is it set later? In that case I would find the method name `set_capacity` confusing.

Reply:
The capacity here is just artificially set by us to bound the queue's size -- we don't really care about what the underlying priority queue's capacity actually is, and I don't even think we can/should interfere with that

Reply:
Oh interesting, hmmm isn't the standard approach to set and bound the capacity of the actual queue at init time and let the queue handle scenarios if its full. That's basically what we now do for the Falco outputs queue. WDYT?

Reply:
Yeah I agree -- the only problem is that the queue is owned by the inspector currently, so in order for the value to be expressed at construction time we'd need a mechanism for either passing the value at the inspector init-time (a bit overkill) or for swapping the queue with another one. If we get to that, I'd rather prefer having a queue that allows resizing at runtime. It's not much cost in terms of synchronization, plus in the real world we'll always resize it when empty anyways (e.g. before opening the inspector).

Reply:
> I'd rather prefer having a queue that allows resizing at runtime

If that's a requirement, perhaps leave a note in the code base.